tests/framework/static_analysis.py

```diff
@@ -0,0 +1,596 @@
+
+
+class X86_64(Architecture):  # pylint: disable=invalid-name
+    """x86_64 architecture"""
+
+    @property
+    def syscall_nr_register(self):
+        return "%eax"
+
+    @property
+    def syscall_argument_registers(self):
+        return ["%rdi", "%rsi", "%rdx", "%r10", "%r8", "%r9"]
+
+    @property
+    def fn_call_argument_registers(self):
+        return ["%rdi", "%rsi", "%rdx", "%rcx", "%r8", "%r9"]
+
+    @property
+    def seccomp_arch(self) -> seccomp.Arch:
+        return seccomp.Arch.X86_64
+
+    def generalize_reg(self, reg: str) -> list[str]:
+        suffixes = ["ax", "bx", "cx", "dx", "si", "di", "bp", "sp"]
+        prefixes = ["%r8", "%r9", "%r10", "%r11", "%r12", "%r13", "%r14", "%r15"]
+
+        for suffix in suffixes:
+            if reg.endswith(suffix):
+                return [f"%r{suffix}", f"%e{suffix}", f"%{suffix}"]
+
+        for prefix in prefixes:
+            if reg.startswith(prefix):
+                return [prefix, f"{prefix}d", f"{prefix}w"]
+
+        return [reg]
+
+    def is_call_instruction(self, insn: Instruction):
+        return insn.mnemonic in ["call", "jmp"]
+
+    def is_syscall_instruction(self, insn: Instruction):
+        return insn.mnemonic == "syscall"
+
+    def backpropagate_register(self, insn: Instruction, reg: str):
+        # Simplifying assumption: an instruction will not modify a register
+        # that it doesn't reference (generally wrong, but fine for our purposes)
+        affected_registers = [
+            match for (match, _) in re.findall(r"(%[a-z0-9]{2,4})(\W|)", str(insn))
+        ]
+        if reg not in affected_registers:
+            return reg
+
+        match insn.mnemonic:
+            case "mov":
+                if len(insn.args) != 2:
+                    raise UnsupportedInstructionException(insn, reg)
+
+                src, dst = insn.args
+
+                if dst == reg:
+                    # an immediate load
+                    if src.startswith("$"):
+                        return int(src[3:], 16)
+                    # We moved something into our target register. If it's a new register, we understand
+                    # what's going on. Anything else, and tough luck
+                    if re.match(r"^%\w{2,4}$", src):
+                        return src
+                    raise UnsupportedInstructionException(insn, reg)
+                return reg
+            case "xor":
+                src, dst = insn.args
+
+                if src == dst:
+                    # we know that reg is part of the arguments, and we know that the arguments are identical
+                    # Thus we have xor reg,reg, which is effectively zeroing reg
+                    return 0
+            case "push":
+                # a push doesn't do anything
+                return reg
+
+        raise UnsupportedInstructionException(insn, reg)
+
+    def instruction_from_str(self, insn_str: str) -> Instruction:
```

Comment:
Sadly it cannot because because the implementations of all these functions are architecture dependent, which is why there's in the `X86_64`/`Aarch64` classes :(